__tests__/app.test.js

```diff
@@ -173,6 +172,176 @@ describe('GET /api/articles/:article_id/comments', () => {
+                username: "rogersop",
+                body: 'this is test for a username missing from a comment'
+            }
+            
+            return request(app)
+            .post(`/api/articles/654654/comments`)
+            .send(newComment)
+            .expect(404)
+            .then(({body}) => {
+                
+                expect(body.message).toBe("Article does not exist")
+        })
+    });
+    test('if there is article_id is not valid responds with 400', () => {
+        const newComment = {
+        username: "rogersop",
+        body: 'this is test for a username missing from a comment'
+    }
+    
+    return request(app)
+    .post(`/api/articles/notID/comments`)
+    .send(newComment)
+    .expect(400)
+    .then(({body}) => {
+        expect(body.message).toBe('Invalid article_id' )
+        })
+    });  
+});
+
+// 8
+describe('PATCH /api/articles/:article_id', () => {
+    test('200: responds with updated article', () => {
+        const articleUpdate = { inc_votes : 1 }
+        return request(app)
+        .patch('/api/articles/1')
+        .send(articleUpdate)
+        .expect(200)
+        .then((response) => {
+            const {article} = response.body;
+            
+            expect(article).toEqual({
+                article_id: 1,
+                title: "Living in the shadow of a great man",
+                topic: "mitch",
+                author: "butter_bridge",
+                body: "I find this existence challenging",
+                created_at: "2020-07-09T20:11:00.000Z",
+                votes: 101,
+                article_img_url: "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+
+            })
+        })
+    });
+    test('200: responds with updated article', () => {
+        const articleUpdate = { inc_votes : -1 }
+        return request(app)
+        .patch('/api/articles/1')
+        .send(articleUpdate)
+        .expect(200)
+        .then((response) => {
+            const {article} = response.body;
+          
+            expect(article).toEqual({
+                article_id: 1,
+                title: "Living in the shadow of a great man",
+                topic: "mitch",
+                author: "butter_bridge",
+                body: "I find this existence challenging",
+                created_at: "2020-07-09T20:11:00.000Z",
+                votes: 99,
+                article_img_url: "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+
+            })
+        })
+    });
+
+    test('404: article doesnt exist', () => {
+        const articleUpdate = { inc_votes : 1 }
+        return request(app)
+        .patch('/api/articles/456454')
+        .send(articleUpdate)
+        .expect(404)
+        .then(({body}) => {
+             
+            expect(body.message).toEqual('Article does not exist')   
+        })
+    });
```

Comment:
Just one test missing here:
Status 400, invalid article id (EG /api/articles/not-a-number)